specification/azsadmin/resource-manager/compute/Microsoft.Compute.Admin/preview/2018-07-30-preview/Disks.json

```diff
@@ -0,0 +1,319 @@
+    "application/json"
+  ],
+  "paths": {
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Compute.Admin/locations/{location}/admindisks": {
+      "get": {
+        "x-ms-examples": {
+          "Returns a list of disks.": {
+            "$ref": "./examples/Disks/List.json"
+          }
+        },
+        "tags": [
+          "Disks"
+        ],
+        "operationId": "Disks_List",
+        "description": "Returns a list of disks.",
+        "parameters": [
+          {
+            "$ref": "../2015-12-01-preview/Compute.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../2015-12-01-preview/Compute.json#/parameters/LocationParameter"
+          },
+          {
+            "$ref": "Disks.json#/parameters/TenantSubscriptionIdParameter"
+          },
+          {
+            "$ref": "Disks.json#/parameters/DiskStateParameter"
+          },
+          {
+            "$ref": "Disks.json#/parameters/SourceSharePathParameter"
+          },
+          {
+            "$ref": "Disks.json#/parameters/DiskCountParameter"
+          },
+          {
+            "$ref": "Disks.json#/parameters/DiskStartParameter"
+          },
+          {
+            "$ref": "Disks.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK -- The list of disks has been returned.",
+            "schema": {
+              "$ref": "Disks.json#/definitions/DiskList"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": null
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Compute.Admin/locations/{location}/admindisks/{DiskId}": {
+      "get": {
+        "x-ms-examples": {
+          "Returns the requested disk.": {
+            "$ref": "./examples/Disks/Get.json"
+          }
+        },
+        "tags": [
+          "Disks"
+        ],
+        "operationId": "Disks_Get",
+        "description": "Returns the disk.",
+        "parameters": [
+          {
+            "$ref": "../2015-12-01-preview/Compute.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../2015-12-01-preview/Compute.json#/parameters/LocationParameter"
+          },
+          {
+            "$ref": "Disks.json#/parameters/DiskIdParameter"
+          },
+          {
+            "$ref": "Disks.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK -- The disk has been returned.",
+            "schema": {
+              "$ref": "Disks.json#/definitions/Disk"
+            }
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "DiskProperties": {
+      "description": "Disk properties.",
+      "type": "object",
+      "properties": {
+        "diskId": {
+          "description": "The disk id.",
+          "type": "string",
+          "readOnly": false
```

Comment:
What happens if I change the disk id?

Reply:
The diskId is what the user specify which disk to operation. If user change the diskId, it means it is operating on different disk, which is by design.

Reply:
I think you are confusing this with something else.  This is the object that is returned.   I think this should be marked as readOnly: true.

Reply:
DiskId cannot be marked as readonly because we need it back to server.

Reply:
Since you are going to change disk migration to an action this should be okay for now.  In the future you should create an <My_Operation>Parameters object which takes all the sufficient and necessary properties to do the operation.  So in this case you would have had an object called DiskMigrationParameters which take in the Disk resource Id and target directory.